src/storage/v2/indices/vector_index.cpp

```diff
@@ -65,9 +65,12 @@ static const std::unordered_map<unum::usearch::metric_kind_t, std::function<doub
 /// The `Impl` structure follows the PIMPL (Pointer to Implementation) idiom to separate
 /// the interface of `VectorIndex` from its implementation
 struct VectorIndex::Impl {
+  /// Maps the index name to the associated IndexItem
+  std::map<std::string, IndexItem> index_;
+
   /// The `index_` member is a map that associates a `LabelPropKey` (a combination of label and property)
-  /// with the pair of a IndexItem.
-  std::map<LabelPropKey, IndexItem> index_;
+  /// with the indexes associated with it
+  std::map<LabelPropKey, std::vector<std::string>> label_prop_to_index_names_;
```

Comment:
I think this is not modeled in the best way. You already have index_name stored under VectorIndexSpec.

Just use
std::map<LabelPropKey, std::vector<IndexItem>>